utils/utils.jsx

```diff
@@ -1066,6 +1066,158 @@ export function getCaretPosition(el) {
     return 0;
 }
 
+export function createHtmlElement(el) {
+    return document.createElement(el);
+}
+
+export function getElementComputedStyle(el) {
+    return getComputedStyle(el);
+}
+
+export function addElementToDocument(el) {
+    document.body.appendChild(el);
+}
+
+export function copyTextAreaToDiv(textArea) {
+    if (!textArea) {
+        return null;
+    }
+    const copy = createHtmlElement('div');
+    copy.textContent = textArea.value;
+    const style = getElementComputedStyle(textArea);
+    [
+        'fontFamily',
+        'fontSize',
+        'fontWeight',
+        'wordWrap',
+        'whiteSpace',
+        'borderLeftWidth',
+        'borderTopWidth',
+        'borderRightWidth',
+        'borderBottomWidth',
+        'paddingRight',
+        'paddingLeft',
+    ].forEach((key) => {
+        copy.style[key] = style[key];
+    });
+    copy.style.overflow = 'auto';
+    copy.style.width = textArea.offsetWidth + 'px';
+    copy.style.height = textArea.offsetHeight + 'px';
+    copy.style.position = 'absolute';
+    copy.style.left = textArea.offsetLeft + 'px';
+    copy.style.top = textArea.offsetTop + 'px';
+    addElementToDocument(copy);
+    return copy;
+}
+
+export function convertRemToPixels(rem) {
+    if (isNaN(rem)) {
+        return 0;
+    }
+    const styles = getElementComputedStyle(document.documentElement);
+    return rem * parseFloat(styles.fontSize);
+}
+
+export function getCaretXYCoordinate(textArea) {
+    if (!textArea) {
+        return {x: 0, y: 0};
+    }
+    const start = textArea.selectionStart;
+    const end = textArea.selectionEnd;
+    const copy = copyTextAreaToDiv(textArea);
```

Comment:
while researching for the best solution to get the X,Y coordinates of the caret, I found several posts in different places that pointed at this solution of creating a div copy with the inner text and some styles from the textArea, as the best available solution. The other potential solution was to use Canvas + 2d and get the width, but that one was expensive in terms of performance. I can take a look at how measuring the selection range behaves for this particular situation.

Reply:
Yeah, I'd imagine that would be more expensive since it's having to set up a different rendering context for that. This solution makes sense to me though.

I wonder if we should look into it for the autosizing textarea in the future since the current method of having an invisible textarea with the same classes causes issues due to CSS getting out of sync occasionally. Something like this would help prevent that.